extmod/mbedtls/mbedtls_config_common.h

```diff
@@ -107,6 +107,11 @@ void m_tracked_free(void *ptr);
 
 #endif
 
+// Include board-level config fragment.
+#ifdef MBEDTLS_BOARD_CONFIG_H
+#include MBEDTLS_BOARD_CONFIG_H
+#endif
```

Comment:
> Why not put this at the start of this file?

Because here it allows you to `#undef` options in common file if needed. For example, I need to `#undef MBEDTLS_SHA512_C` this when enabling crypto device.

> But actually I think it would be better to just let the board override MBEDTL_CONFIG_FILE, and specify its own entire config file

Well the board mk/cmake is always included first, wouldn't it be enough to check if `MBEDTLS_CONFIG_FILE` is defined/set ?

Reply:
That may work, but it'll be rather ugly.  The board would do `CFLAGS += -DMBEDTLS_CONFIG_FILE` and then the main makefile would need to check `CFLAGS` and if it doesn't contain the definition, then add it.  That's not a great solution because it relies heavily on the order of defining CFLAGS.

I thing having a `CFLAG_BOARD` which is added at the very end of `CFLAGS` would be the most general thing to do, and help other cases where a board may want to override settings.

Reply:
> But actually I think it would be better to just let the board override MBEDTL_CONFIG_FILE

This is implemented now, although I think the old solution might have been the cleanest, and it allowed defining/undef'ing options just before the check config include in the common config file.

Reply:
I think including the port's mbedtls_config.h should be moved to `extmod.mk` for Makefile-based ports too.

Reply:
For the above I meant in a follow-up PR, that I can send after this is merged (or do you want this now ? fine either way) It would be a nice to make `extmod.mk` include the port's default mbedtls_config.h if Not set, just like `extmod.mk` does right now, this way you can override it from any port.

Reply:
> This is implemented now

Thanks, that looks pretty good now!

> lthough I think the old solution might have been the cleanest, and it allowed defining/undef'ing options just before the check config include in the common config file

That's true... but I think you can just include the check config again after the board made all its changes (you'll need to do `#undef MBEDTLS_CHECK_CONFIG_H` first though).

Reply:
> For the above I meant in a follow-up PR, that I can send after this is merged (or do you want this now ? fine either way) It would be a nice to make extmod.mk include the port's default mbedtls_config.h if Not set, just like extmod.mk does right now, this way you can override it from any port.

Yes I saw that, would be good to move the config to `extmod.mk` (like FFCONF_H is there).  You can do that in a separate PR.

Reply:
#9899